vendor-extra/LiberoPageBundle/src/Event/LoadPageEvent.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\LiberoPageBundle\Event;
+
+use GuzzleHttp\Promise\PromiseInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\HttpFoundation\Request;
+use function GuzzleHttp\Promise\all;
+
+final class LoadPageEvent extends Event
+{
+    public const NAME = 'libero.page.load';
+
+    private $context;
+    private $documents = [];
+    private $request;
+
+    public function __construct(Request $request, array $context = [])
+    {
+        $this->request = $request;
+        $this->context = $context;
+    }
+
+    public function getRequest() : Request
+    {
+        return $this->request;
+    }
+
+    public function addDocument(string $key, PromiseInterface $promise) : void
+    {
+        $this->documents[$key] = $promise;
+    }
+
+    public function getDocuments() : PromiseInterface
+    {
+        return all($this->documents);
```

Comment:
This will be problematic when handling lists: there will be follow-up requests for each item, which won't be included here.

Reply:
Wonder if they can be the same event? 🤔

Reply:
```php
return all($this->documents)->then(
    function () : PromiseInterface {
        return all($this->documents);
    }
);
```

looks weird, but might work?

Reply:
Thinking about this, we probably don't want to be eager-loading everything. This would result in a lot of chained requests (eg list > item > item categories > category). HTTP caching helps, but avoiding the requests entirely would be better (ie caching the thing that would be produced, like the teaser).

Reply:
Done a quick experiment and seems to work. Essentially needs replacing `View` with one that wraps the HTTP request and conversion. https://github.com/libero/browser/blob/5d8b471cab69b20b93997291ef41d46722eb22e1/vendor-extra/ViewsBundle/src/Views/ViewConverterRegistry.php#L36-L38 needs consideration as accessing details should unwrap it, so need a way to stop propagation (special context key?).

Reply:
This is demoed in #56.